lib/modules/manager/gradle/extract.ts

```diff
@@ -137,12 +137,18 @@ export async function extractAllPackageFiles(
         };
       }
 
-      dep.registryUrls = getRegistryUrlsForDep(packageRegistries, dep);
+      if (!dep.datasource) {
+        dep.datasource = datasource;
+      }
+
+      if (dep.datasource === datasource) {
```

Comment:
I think renaming `datasource` to `mavenDatasource` now makes sense, if we're about to also use `npm` soon. Also, I'm okay with doing it later.